lib/parsers/gjs-parser.js

```diff
@@ -0,0 +1,430 @@
+        (t) => t.replacedRange[0] === range[0] && t.replacedRange[1] === range[1]
+      );
+      if (!template) {
+        return null;
+      }
+      counter++;
+      const ast = template.ast;
+      Object.assign(node, ast);
+    }
+
+    if ('blockParams' in node) {
+      const upperScope = findParentScope(result.scopeManager, path);
+      const scope = result.isTypescript
+        ? new TypescriptScope.BlockScope(result.scopeManager, upperScope, node)
+        : new Scope(result.scopeManager, 'block', upperScope, node);
+      for (const [i, b] of node.params.entries()) {
+        const v = new Variable(b.name, scope);
+        v.identifiers.push(b);
+        v.defs.push(new Definition('Parameter', b, node, node, i, 'Block Param'));
+        scope.variables.push(v);
+        scope.set.set(b.name, v);
+      }
+    }
+
+    if (node.type === 'GlimmerPathExpression' && node.head.type === 'VarHead') {
+      const name = node.head.name;
+      if (glimmer.isKeyword(name)) {
+        return null;
+      }
+      const { scope, variable } = findVarInParentScopes(result.scopeManager, path, name) || {};
+      if (scope) {
+        node.head.parent = node;
+        registerNodeInScope(node.head, scope, variable);
+      }
+    }
+    if (node.type === 'GlimmerElementNode') {
+      node.name = node.tag;
+      const { scope, variable } = findVarInParentScopes(result.scopeManager, path, node.tag) || {};
+      if (scope && (variable || isUpperCase(node.tag[0]))) {
+        registerNodeInScope(node, scope, variable);
+      }
+    }
+    return null;
+  });
+
+  if (counter !== templateInfos.length) {
+    throw new Error('failed to process all templates');
+  }
+}
+
+/**
+ * implements https://eslint.org/docs/latest/extend/custom-parsers
+ * 1. transforms gts/gjs files into parseable ts/js without changing the offsets and locations around it
+ * 2. parses the transformed code and generates the AST for TS ot JS
+ * 3. preprocesses the templates info and prepares the Glimmer AST
+ * 4. converts the js/ts AST so that it includes the Glimmer AST at the right locations, replacing the original
+ */
+module.exports = {
+  parseForESLint(code, options, isTypescript) {
+    let jsCode = code;
+    const info = gts.transformForLint({
+      input: jsCode,
+      templateTag: 'template',
+      explicitMode: true,
+      linterMode: true,
+    });
+    jsCode = info.output;
+
+    let result = null;
+    result = isTypescript
+      ? typescriptParser.parseForESLint(jsCode, { ...options, ranges: true })
+      : babelParser.parseForESLint(jsCode, { ...options, ranges: true });
+    if (!info.replacements?.length) {
+      return result;
+    }
+    const preprocessedResult = preprocessGlimmerTemplates(info, code);
+    const { templateVisitorKeys } = preprocessedResult;
+    const visitorKeys = { ...result.visitorKeys, ...templateVisitorKeys };
+    result.isTypescript = isTypescript;
+    convertAst(result, preprocessedResult, visitorKeys);
```

Comment:
does this mean that the template-lint rules could be added as eslint rules? I saw you had `@glimmer/syntax` as well?

Reply:
yes, I'm already having most of them working with this parser over at eslint-plugin-template-lint. running them natively.